### DISCUSSION TITLE
[QUESTION]: Pass ENV variables from server to client without replacing in the code

### DISCUSSION THREAD

**Discussion by simplecommerce** [vikejs/vike#1487] [Discussion] [answered]
Hi,

I have a quick question.
I was using `import.meta.env` to pass some variables from server to client in my test project.
But I just noticed that when building for production, it actually replaces the line instead of taking it from the ENV variable.
I saw the line in your guide afterwards.

So now I am wondering, if I wanted to pass process.env.ROOT_URL (example) to my client side but not have it bundled in the production and actually take it from the actual environment variable, how would I do this?

I know I can inject html from the server to the client side and pick it up later this way.
I don't think I can use passToClient because the value I need to pass isn't used in a page I am rendering but in a function call.

Sorry if I am not clear.
I was just wondering how you guys do it.

Thanks.

**Comment by brillout** [comment, marked as answer]
If your en var is expected to be determined/changed dynamically at runtime, then you'll indeed have to use `pageContext` and `passToClient`. As described in the Data Fetching guide.
